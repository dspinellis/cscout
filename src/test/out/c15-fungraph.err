Processing workspace TestWS
Processing directory test/c
Entering directory test/c
Processing project Prj1
Processing file c15-fungraph.c
/cscout/src/test/c/c15-fungraph.c:9: warning: undeclared identifier: function3
The text leading to this warning is: [function3(]
/cscout/src/test/c/c15-fungraph.c:9: warning: assuming declaration int function3(...)
The text leading to this warning is: [function3()]
/cscout/src/test/c/c15-fungraph.c:10: warning: undeclared identifier: printf
The text leading to this warning is: [printf(]
/cscout/src/test/c/c15-fungraph.c:10: warning: assuming declaration int printf(...)
The text leading to this warning is: [printf("Hello")]
Done processing file c15-fungraph.c
Done processing project Prj1
Processing project Prj2
Processing file c15-fungraph.c
/cscout/src/test/c/c15-fungraph.c:9: warning: undeclared identifier: function3
The text leading to this warning is: [function3(]
/cscout/src/test/c/c15-fungraph.c:9: warning: assuming declaration int function3(...)
The text leading to this warning is: [function3()]
/cscout/src/test/c/c15-fungraph.c:10: warning: undeclared identifier: printf
The text leading to this warning is: [printf(]
/cscout/src/test/c/c15-fungraph.c:10: warning: assuming declaration int printf(...)
The text leading to this warning is: [printf("Hello")]
Done processing file c15-fungraph.c
Processing file prj2.c
Done processing file prj2.c
Done processing project Prj2
Exiting directory test/c
Done processing directory test/c
Done processing workspace TestWS
Post-processing /cscout/example/.cscout/host-defs.h
Post-processing /cscout/example/.cscout/host-incs.h
Post-processing /cscout/src/makecs.cs
Post-processing /cscout/src/test/c/c15-fungraph.c
Post-processing /cscout/src/test/c/prj2.c
Processing identifiers
5%10%15%20%25%30%35%40%45%50%55%60%65%70%75%80%85%90%95%100%
