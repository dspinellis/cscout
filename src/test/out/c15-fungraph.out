Table: Ids
EID	NAME	READONLY	UNDEFMACRO	MACRO	FUNMACRO	MACROARG	CPPCONST	CPPSTRVAL	DEFCCONSTVAL	NOTDEFCCONSTVAL	EXPCCONSTVAL	NOTEXPCCONSTVAL	ORDINARY	SUETAG	SUMEMBER	LABEL	TYPEDEF	ENUM	YACC	FUN	CSCOPE	LSCOPE	UNUSED
89	macro	0	0	1	1	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0
91	MAXDIGIT	0	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
169	middlemacro	0	0	1	1	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0
321	x	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
421	main	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	0
529	innemacro	0	0	1	1	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0
541	qqq	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
821	foo	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
849	function1	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	0
1119	function2	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	0
1819	function3	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
2029	printf	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
2975	__DATE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3295	__TIME__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3575	__FILE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3865	__LINE__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
4055	__STDC__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
5355	_cscout_dummy1	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	1	0	0
Table: Tokens
FID	FOFFSET	EID
2	297	2975
2	329	3295
2	357	3575
2	386	3865
2	405	4055
2	471	421
2	535	5355
2	558	5355
4	8	89
4	16	169
4	38	169
4	52	529
4	72	529
4	84	849
4	96	849
4	111	1119
4	126	421
4	143	89
4	160	1119
4	181	1819
4	202	2029
5	8	91
5	31	321
5	41	421
5	53	541
5	81	821
Table: Rest
FID	FOFFSET	CODE
2	287	\u0000a\u0000a#define 
2	305	  
2	320	\u0000a#define 
2	337	 
2	348	\u0000a#define 
2	365	 
2	377	\u0000a#define 
2	394	 1\u0000a#define 
2	413	 1\u0000a\u0000a
2	466	\u0000aint 
2	475	();\u0000a
2	522	\u0000astatic void 
2	549	(void) { 
2	572	(); }\u0000a
3	152	\u0000a\u0000a\u0000a#pragma includepath 
3	195	\u0000a
3	239	\u0000astatic void _cscout_dummy2(void) { _cscout_dummy2(); }\u0000a
4	0	#define 
4	13	() 
4	27	()\u0000a#define 
4	49	() 
4	61	()\u0000a#define 
4	81	() 
4	93	()\u0000a
4	105	() {}\u0000a
4	120	() {}\u0000a
4	130	() {\u0000a        
4	148	();\u0000a        
4	169	();\u0000a        
4	190	();\u0000a        
4	208	(
4	216	);\u0000a}\u0000a
5	0	#define 
5	16	 11\u0000aextern int 
5	32	;\u0000aextern 
5	45	();\u0000aint 
5	56	;\u0000a\u0000a
5	80	\u0000a
5	84	() {\u0000a	
5	107	}\u0000a\u0000a
Table: Projects
PID	NAME
23	unspecified
24	Prj1
25	Prj2
Table: IdProj
EID	PID
89	24
169	24
421	24
529	24
849	24
1119	24
1819	24
2029	24
2975	24
3295	24
3575	24
3865	24
4055	24
5355	24
89	25
91	25
169	25
321	25
421	25
529	25
541	25
821	25
849	25
1119	25
1819	25
2029	25
2975	25
3295	25
3575	25
3865	25
4055	25
5355	25
Table: Files
FID	NAME	RO
2	host-defs.h	1
3	host-incs.h	1
4	c15-fungraph.c	0
5	prj2.c	0
Table: Filemetrics
FID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NCOPIES	NINCFILE	NPFUNCTION	NFFUNCTION	NPVAR	NFVAR	NAGGREGATE	NAMEMBER	NENUM	NEMEMBER
2	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	16	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	2	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	1	0	0	0	0	0	0
2	1	548	367	29	0	3	22	61	47	NULL	1	1	0	0	0	5	0	0	5	37	2	0	0	2	0	3	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	2	5	8	1	1	5	7	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
3	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	15	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	0	0	0	0	0	0	0
3	1	243	187	13	0	2	13	48	28	NULL	1	1	0	0	0	1	0	0	0	16	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
4	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	24	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	33	4	0	0	0	0	1	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	14	0	NULL	14	5	0	NULL	5	0	NULL	NULL	NULL	3	0	0	0	0	0	0	0
4	1	219	0	52	0	0	11	30	58	NULL	1	1	0	0	0	3	0	3	0	56	4	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	7	0	6	13	5	0	3	8	0	4	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
5	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	7	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	3	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	NULL	4	4	0	NULL	4	0	NULL	NULL	NULL	1	0	2	0	0	0	0	0
5	1	95	32	17	2	0	11	20	13	NULL	1	1	0	0	0	1	0	0	1	21	3	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	1	5	4	0	1	5	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
Table: FileProj
FID	PID
2	24
3	24
4	24
1	25
2	25
3	25
4	25
5	25
Table: LineProj
FID	PID	LNUM
2	24	1
2	24	2
2	24	3
2	24	4
2	24	5
2	24	6
2	24	7
2	24	8
2	24	9
2	24	10
2	24	11
2	24	12
2	24	13
2	24	14
2	24	15
2	24	16
2	24	17
2	24	18
2	24	19
2	24	20
2	24	21
2	24	22
3	24	1
3	24	2
3	24	3
3	24	4
3	24	5
3	24	6
3	24	7
3	24	8
3	24	9
3	24	10
3	24	11
3	24	12
3	24	13
4	24	1
4	24	2
4	24	3
4	24	4
4	24	5
4	24	6
4	24	7
4	24	8
4	24	9
4	24	10
4	24	11
2	25	1
2	25	2
2	25	3
2	25	4
2	25	5
2	25	6
2	25	7
2	25	8
2	25	9
2	25	10
2	25	11
2	25	12
2	25	13
2	25	14
2	25	15
2	25	16
2	25	17
2	25	18
2	25	19
2	25	20
2	25	21
2	25	22
3	25	1
3	25	2
3	25	3
3	25	4
3	25	5
3	25	6
3	25	7
3	25	8
3	25	9
3	25	10
3	25	11
3	25	12
3	25	13
4	25	1
4	25	2
4	25	3
4	25	4
4	25	5
4	25	6
4	25	7
4	25	8
4	25	9
4	25	10
4	25	11
5	25	1
5	25	2
5	25	3
5	25	4
5	25	5
5	25	6
5	25	7
5	25	8
5	25	9
5	25	10
5	25	11
Table: LinePos
FID	FOFFSET	LNUM
2	0	1
2	4	2
2	51	3
2	54	4
2	100	5
2	119	6
2	122	7
2	184	8
2	231	9
2	234	10
2	284	11
2	288	12
2	289	13
2	321	14
2	349	15
2	378	16
2	397	17
2	416	18
2	417	19
2	467	20
2	479	21
2	523	22
3	0	1
3	4	2
3	51	3
3	54	4
3	78	5
3	97	6
3	100	7
3	149	8
3	153	9
3	154	10
3	155	11
3	196	12
3	240	13
4	0	1
4	30	2
4	64	3
4	96	4
4	111	5
4	126	6
4	135	7
4	152	8
4	173	9
4	194	10
4	219	11
5	0	1
5	20	2
5	34	3
5	49	4
5	58	5
5	59	6
5	80	7
5	81	8
5	89	9
5	107	10
5	109	11
Table: Definers
PID	CUID	BASEFILEID	DEFINERID
24	4	4	2
25	4	4	2
25	5	5	2
Table: Includers
PID	CUID	BASEFILEID	INCLUDERID
24	2	2	1
24	4	3	1
24	4	4	1
25	2	2	1
25	2	2	1
25	4	3	1
25	4	4	1
25	5	3	1
25	5	5	1
Table: Providers
PID	CUID	PROVIDERID
24	2	2
24	4	4
25	2	2
25	2	2
25	4	4
25	5	5
Table: IncTriggers
PID	CUID	BASEFILEID	DEFINERID	FOFFSET	LEN
24	4	4	2	471	4
25	4	4	2	471	4
25	5	5	2	471	4
Table: Functions
ID	NAME	ISMACRO	DEFINED	DECLARED	FILESCOPED	FID	FOFFSET	FANIN
89	macro	1	1	0	1	4	8	1
389	middlemacro	1	1	0	1	4	38	1
729	innemacro	1	1	0	1	4	72	1
821	foo	0	1	1	0	5	81	0
969	function1	0	1	1	0	4	96	1
1119	function2	0	1	1	0	4	111	1
1269	main	0	1	1	0	4	126	0
1819	function3	0	0	1	0	4	181	1
2029	printf	0	0	1	0	4	202	1
5355	_cscout_dummy1	0	1	1	1	2	535	1
Table: FunctionDefs
FUNCTIONID	FIDBEGIN	FOFFSETBEGIN	FIDEND	FOFFSETEND
89	4	15	4	30
389	4	51	4	64
729	4	83	4	96
821	5	88	5	108
969	4	109	4	110
1119	4	124	4	125
1269	4	134	4	220
5355	2	557	2	577
Table: FunctionMetrics
FUNCTIONID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NGNSOC	NMPARAM	NFPARAM	NEPARAM	FANIN	FANOUT	CCYCL1	CCYCL2	CCYCL3	CSTRUC	CHAL	IFLOW
89	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	0	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
89	1	17	0	2	0	0	1	15	0	NULL	0	1	0	0	0	0	0	0	0	3	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	1	0	0	1	1	0	0	NULL	0	NULL	0	1	1	1	1	1	1.0	0.0	1.0
389	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	0	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
389	1	15	0	2	0	0	1	13	0	NULL	0	1	0	0	0	0	0	0	0	3	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	1	0	0	1	1	0	0	NULL	0	NULL	0	1	1	1	1	1	1.0	0.0	1.0
729	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	0	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
729	1	24	0	2	0	0	1	13	0	NULL	0	1	0	0	0	0	0	0	0	3	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	2	0	0	2	1	0	0	1	0	0	NULL	0	NULL	0	1	1	1	1	1	1.0	0.0	1.0
821	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	1	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	20	NULL	0	NULL	0	0	1	1	1	0.0	0.0	0.0
821	1	21	14	3	1	0	3	17	0	NULL	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	NULL	0	0	0	1	1	1	0.0	0.0	0.0
969	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	1	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	14	NULL	0	NULL	1	0	1	1	1	0.0	0.0	0.0
969	1	2	0	1	0	0	1	1	0	NULL	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	NULL	0	1	0	1	1	1	0.0	0.0	0.0
1119	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	1	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	16	NULL	0	NULL	1	0	1	1	1	0.0	0.0	0.0
1119	1	2	0	1	0	0	1	1	0	NULL	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	NULL	0	1	0	1	1	1	0.0	0.0	0.0
1269	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	6	0	0	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	4	0	0	0	0	1	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	NULL	4	4	0	NULL	4	0	NULL	16	NULL	0	NULL	0	4	1	1	1	0.0	8.0	0.0
1269	1	91	0	38	0	0	6	25	7	NULL	0	1	0	0	0	0	0	0	0	18	4	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	3	0	1	4	3	0	1	4	0	2	NULL	0	NULL	0	0	4	1	1	1	0.0	8.0	0.0
5355	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	0	0	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	NULL	1	0	1	NULL	1	0	NULL	12	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
5355	1	22	0	3	0	0	1	21	5	NULL	0	1	0	0	0	0	0	0	0	5	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	1	0	1	0	1	0	0	NULL	0	NULL	0	1	1	1	1	1	1.0	0.0	1.0
Table: FunctionId
FUNCTIONID	ORDINAL	FID	FOFFSET
89	0	4	8
389	0	4	38
729	0	4	72
821	0	5	81
969	0	4	96
1119	0	4	111
1269	0	2	471
1819	0	4	181
2029	0	4	202
5355	0	2	535
Table: Fcalls
SOURCEID	DESTID
89	389
389	729
729	969
1269	89
1269	1119
1269	1819
1269	2029
5355	5355
Done
